link/link_test.go

```diff
@@ -91,6 +91,16 @@ func mustCgroupFixtures(t *testing.T) (*os.File, *ebpf.Program) {
 	testutils.SkipIfNotSupported(t, haveProgAttach())
 
 	prog := mustCgroupEgressProgram(t)
+	cgroup := mustCgroupFixturesWithProgram(t, prog)
+
+	return cgroup, prog
+}
+
+func mustCgroupFixturesWithProgram(t *testing.T, prog *ebpf.Program) *os.File {
+	t.Helper()
+
+	testutils.SkipIfNotSupported(t, haveProgAttach())
```

Comment:
This skip doesn't make sense since you're splitting program creation out of the helper. Can you move this call into the functions that actually call `NewProgram`?